homeassistant/components/simplisafe/camera.py

```diff
@@ -0,0 +1,234 @@
+            self,
+            simplisafe,
+            system,
+            device=camera,
+            additional_websocket_events=[EVENT_CAMERA_MOTION_DETECTED],
+        )
+        Camera.__init__(self)
+
+        self._hass = hass
+
+    @property
+    def name(self) -> str | None:
+        """Return a good name for this camera."""
+        return self._device.name
+
+    @callback
+    def async_unload(self) -> None:
+        """Release resources."""
+        for service in SERVICES:
+            self._hass.services.async_remove(DOMAIN, service)
+
+    async def async_camera_motion_image(
+        self, width: int | None = None, height: int | None = None
+    ) -> bytes | None:
+        """Return bytes of camera image."""
+        if self._attr_cached_image is not None:
+            return self._attr_cached_image
+
+        if self._attr_image_url is None:
+            return None
+
+        self._attr_cached_image = await self._simplisafe.async_media_request(
+            self._attr_image_url.replace("{&width}", "&width=" + str(width))
+        )
+        return self._attr_cached_image
+
+    async def async_save_image_handler(self, filename: Template, width: int) -> None:
+        """Handle the service call to save a motion image."""
+        if self._attr_image_url is None:
+            return
+
+        filename.hass = self._hass
+        snapshot_file: str = filename.async_render(
+            variables={"entity_id": self.entity_id}
+        )
+
+        try:
+            snapshot = await self._simplisafe.async_media_request(
+                self._attr_image_url.replace("{&width}", "&width=" + str(width))
+            )
+        except SimplipyError as err:
+            raise HomeAssistantError(
+                f'Error fetching motion media "{self._system.system_id}": {err}'
+            ) from err
+
+        try:
+            await self._hass.async_add_executor_job(
+                _write_image, snapshot_file, snapshot
+            )
+        except OSError as err:
+            LOGGER.error("Can't write image to file: %s", err)
+
+    async def async_save_clip_handler(self, filename: Template) -> None:
+        """Handle the service call to save a motion clip."""
+        if self._attr_clip_url is None:
+            return
+
+        filename.hass = self._hass
+        clip_file: str = filename.async_render(variables={"entity_id": self.entity_id})
+
+        try:
+            clip = await self._simplisafe.async_media_request(self._attr_clip_url)
+        except SimplipyError as err:
+            raise HomeAssistantError(
+                f'Error fetching motion media "{self._system.system_id}": {err}'
+            ) from err
+
+        try:
+            await self._hass.async_add_executor_job(_write_image, clip_file, clip)
+        except OSError as err:
+            LOGGER.error("Can't write image to file: %s", err)
+
+    @callback
+    def async_update_from_websocket_event(self, event: WebsocketEvent) -> None:
```

Comment:
`EVENT_CAMERA_MOTION_DETECITED` is not included in `WEBSOCKET_EVENTS_REQUIRING_SERIAL`, does it mean every camera will trigger when there's a motion event?